src/main/scala/chisel3/simulator/Simulator.scala

```diff
@@ -54,6 +54,31 @@ trait Simulator[T <: Backend] {
   def commonCompilationSettings:          CommonCompilationSettings
   def backendSpecificCompilationSettings: backend.CompilationSettings
 
+  /** Post process a simulation log to see if there are any backend-specific failures.
+    *
+    * @return None if no failures found or Some if they are
+    */
+  private def postProcessLog: Option[Throwable] = {
+    val log = Paths.get(workspacePath, s"workdir-${tag}", "simulation-log.txt").toFile
+    val lines = scala.io.Source.fromFile(log).getLines().zipWithIndex.filter { case (line, lineNo) => backend.assertionFailed.matches(line) }.toSeq
+    if (lines.isEmpty)
+      return None
+
+    Some(
```

Comment:
```suggestion
    Option.when(lines.nonEmpty)(
```
I'm not against return but it feels egregious here 😉